runtime/server/auth/jwt.go

```diff
@@ -0,0 +1,223 @@
+package auth
+
+import (
+	"context"
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/MicahParks/keyfunc"
+	"github.com/golang-jwt/jwt/v4"
+	"go.uber.org/zap"
+	"gopkg.in/square/go-jose.v2"
+)
+
+// Issuer creates JWTs with claims for an Audience.
+// The Issuer is used by the admin server to create JWTs for the runtimes it manages based on a user's control-plane permissions.
+type Issuer struct {
```

Comment:
It will be imported/used by the admin server, since it's acting as an issuer of runtime tokens, but I think it makes sense to implement it in the runtime. It's also simpler to group it with the validation logic (like for the testing).